include/okapi/impl/device/opticalSensor.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "api.h"
+#include "okapi/api/control/controllerInput.hpp"
+#include "okapi/api/filter/passthroughFilter.hpp"
+#include <memory>
+
+namespace okapi {
+class OpticalSensor : public ControllerInput<double> {
+  public:
+  /**
+   * An optical sensor on a V5 port.
+   *
+   * ```cpp
+   * auto os = OpticalSensor(1);
+   * ```
+   *
+   * @param iport The V5 port the device uses
+   */
+  OpticalSensor(std::uint8_t iport);
+
+  virtual ~OpticalSensor();
+
+  /**
+   * Returns the current sensor's hue value (0-359.99).
+   *
+   * @return current value
+   */
+  virtual double get();
+
+  /**
+   * Returns the current sensor's brightness value (0-1.0).
+   *
+   * @return current value
+   */
+  virtual double get_brightness();
+
+  /**
+   * Returns the current sensor's saturation value (0-1.0).
+   *
+   * @return current value
+   */
+  virtual double get_saturation();
+
+  /**
+   * Returns the current sensor's LED brightness
+   *
+   * value that ranges from 0 to 100
+   *
+   * @return current value
```

Comment:
```suggestion
   * @return The current sensor value in the range [0, 100].
```
Then you can delete the middle two lines of the comment.

Same with below.